api/routes/deleteEntries.js

```diff
@@ -1,17 +1,19 @@
+//delete an entry by id
 const express = require('express')
 const route = express.Router()
 const database = require('../models/post.model')
 
 route.delete('/:id', (req, res) =>{
-    let match = false
-    const {id} = req.params
+let match = false
+const {id} = req.params
 for(let i = 0; i < database.Posts.length; i++){
```

Comment:
Is your code doing an extra work here? Observe that you don't need to loop through the array to delete an item. This endpoint still needs some thoughts from you.

if I have posts of id 0 to 9. If I delete id=5 (which is on array index = 5), the item with id=6, will now be on array index 5 and  id=7 item which would now have array index 6. So if I try to delete id=6 and the endpoint does this: Posts.splice(6, 1), you would remove the id=7 item because it is on array index 6.

Reply:
I will make the changes . Thank you Theo